api/client/utils.go

```diff
@@ -94,13 +94,14 @@ func (cli *DockerCli) clientRequest(method, path string, in io.Reader, headers m
 	if resp != nil {
 		serverResp.statusCode = resp.StatusCode
 	}
+
 	if err != nil {
-		if strings.Contains(err.Error(), "connection refused") {
+		if strings.Contains(err.Error(), "connection refused") || strings.Contains(err.Error(), "dial unix") || strings.Contains(err.Error(), "i/o timeout") {
```

Comment:
That's the go http package that does that. The URL itself isn't what is causing the connection issue, it's that literally the unix sock isn't there.